episodes/filters-and-thresholding.md

```diff
@@ -0,0 +1,743 @@
+`Plugins > napari Matplotlib > Histogram`
+
+![](fig/blurred-nuclei-histogram.png){alt="A histogram of the 29th z slice of 
+the nuclei image after a gaussian blur. The left contrast limit is set 
+to 0.134."}
+
+Make sure you have 'Result of gaussian_blur' selected in the layer list (should 
+be highlighted in blue).
+
+You should see that this histogram now runs from 0 to 1, reflecting the new 
+values after the gaussian blur and its new data type. Let's again choose a 
+threshold between the two peaks - around 0.134:
+
+```python
+
+# Get the image data for the blurred nuclei
+blurred = viewer.layers["Result of gaussian_blur"].data
+
+# Create mask with a threshold of 0.134
+blurred_mask = blurred > 0.134
+
+# Add mask to the Napari viewer
+viewer.add_labels(blurred_mask)
+
+```
+
+![](fig/threshold-blurred-mask.png){alt="Mask of nuclei (brown) overlaid on 
+nuclei image - created with manual thresholding after gaussian blur"}
+
+Now we see a much more sensible result - a brown mask that again highlights the 
+nuclei in brown. If we compare to the previous mask, we can see that this one 
+seems much smoother and has fewer areas inside nuclei that aren't labelled, as 
+well as fewer areas of the background that are incorrectly labelled. 
+
+This workflow of blurring an image then thresholding it is a very common method 
+to provide a smoother, more complete mask. Note that it still isn't perfect! If 
+you browse through the z slices with the slider at the bottom of the image, it's 
+clear that some areas are still missed or incorrectly labelled.
+
+## Automated thresholding
+
+First, let's clean up our layer list again. Make sure you only have the 'nuclei' 
+and 'Result of gaussian_blur' layers in the layer list - select any others and 
+remove them by clicking the ![](
+https://raw.githubusercontent.com/napari/napari/main/napari/resources/icons/delete.svg
+){alt="A screenshot of Napari's delete layer button" height='30px'} icon. Then, 
+if you still have the `napari-matplotlib` histogram open, close it by clicking 
+the tiny `x` icon in the top left corner.
+
+So far we have chosen our threshold manually by looking at the image histogram, 
+but it would be better to find an automated method to do this for this. Many 
+such methods exist - for example, see the various options starting with 
+'Threshold' under `Tools > Segmentation / binarization`. One of the most common 
+methods is _Otsu thresholding_, which we will look at now.
+
+Let's go ahead and apply this to our blurred image:  
+`Tools > Segmentation / binarization > Threshold (Otsu et al 1979, scikit-image, nsbatwm)`
+
+This will open a panel on the right side of Napari. Select 'Result of 
+gaussian_blur(data)' in the image row, then click Run:
+
+![](fig/threshold-blurred-otsu-mask.png){alt="Mask of nuclei (brown) overlaid on 
+nuclei image - created with Otsu thresholding after gaussian blur"}
+
+This should produce a mask that is almost identical to the one we created with a 
```

Comment:
From my eye, it looks like Otsu is a bit more conservative on where the boundary ended compared to the manual threshold, but that may be me playing with the contrast limits.

I actually wonder maybe we should ask them to keep some of the mask layers so that they can compare. Perhaps show them how to rename the layers so that they are called "Mask-Manual", "Mask-Manual Smoothed" "Mask-Otsu Smoothed" so that they can toggle them on and off and compare.